web-app/src/components/HolidayModal/container.js

```diff
@@ -11,19 +15,70 @@ export default Wrapped =>
 
     constructor(props) {
       super(props);
-      this.state = {};
+      this.state = {
+        holiday: {},
+      };
     }
 
-    approveHoliday = holidayId => {
-      console.log('approved');
+    componentWillUpdate(nextProps) {
+      if (nextProps.holiday !== this.props.holiday) {
+        this.setState({ holiday: nextProps.holiday });
+      }
+    }
+
+    approveHoliday = () => {
+      const holidayId = this.state.holiday.holidayId;
+
+      approveHoliday(holidayId)
+        .then(() => {
+          this.setState({
+            holiday: {
+              ...this.state.holiday,
+              eventStatus: {
+                ...this.state.holiday.eventStatus,
+                eventStatusId: holidayStatus.APPROVED,
+              },
+            },
+          });
+          Toast({
+            type: 'success',
+            title: 'Holiday Approved',
+            position: 'top-end',
+          });
+        })
+        .catch(error => {
+          swal('Error', `There was an error: ${error.message}. 'error`);
+        });
     };
 
-    rejectHoliday = holidayId => {
-      console.log('rejected');
+    rejectHoliday = () => {
+      const holidayId = this.state.holiday.holidayId;
+
+      rejectHoliday(holidayId)
```

Comment:
The two rejectHoliday methods (one class and the other for services) confused me. Maybe consider renaming the class method to onRejectHoliday?